cocos/2d/CCSpriteFrameCache.cpp

```diff
@@ -60,6 +60,14 @@ SpriteFrameCache* SpriteFrameCache::getInstance()
     return _sharedSpriteFrameCache;
 }
 
+void SpriteFrameCache::setDelegate(SpriteFrameCache* pCache)
+{
+    swap(_sharedSpriteFrameCache, pCache);
+    if (pCache){
```

Comment:
`CC_SAFE_RELEASE_NULL` tests the pointer for nullness so it's useless to test it here.